CMakeLists.txt

```diff
@@ -87,15 +81,15 @@ if(OTF2XX_WITH_MPI)
         INTERFACE
             OTF2XX_HAS_MPI
     )
+    find_package(Boost REQUIRED COMPONENTS mpi)
+    find_package(MPI REQUIRED COMPONENTS CXX)
+    if(NOT TARGET MPI::MPI_CXX)
+        message(FATAL_ERROR "MPI support requires CMake >= 3.9 or a properly defined MPI::MPI_CXX target")
+    endif()      
     target_link_libraries(otf2xx-core INTERFACE Boost::mpi MPI::MPI_CXX)
 endif()
 
-set(READER_SRCS src/reader/callback/definitions.cpp src/reader/callback/events.cpp)
-if(OTF2XX_USE_STATIC_LIBS)
```

Comment:
When I'm thinking about it, it's good that the shared build never worked, as the shared library will not be installed.

Reply:
Why? You install the target and if the target is a shared lib it will be installed. Or am I missing anything?

Reply:
If you have otf2xx as a git submodule using add_subdirectory(), then there is no install().

Reply:
Yes. But in that case the consumer is responsible for making sure the library exists (however he does that), when building and installing OTF2xx everything is installed correctly which is IMO the only important thing here.